File: 2024/dec04/02.py
import xmas

XmasCount=0

for linh in range(1, xmas.howManyLines-1):
    for coln in range(1, xmas.howManyCols-2): #eliminar o \n
        if xmas.lines[linh][coln] == "A":
            XmasCount+=1 if xmas.x_mas(xmas.lines, linh, coln) == ("MAS", "MAS") else 0
            XmasCount+=1 if xmas.x_mas(xmas.lines, linh, coln) == ("MAS", "SAM") else 0
            XmasCount+=1 if xmas.x_mas(xmas.lines, linh, coln) == ("SAM", "MAS") else 0
            XmasCount+=1 if xmas.x_mas(xmas.lines, linh, coln) == ("SAM", "SAM") else 0

print("X-MAS Count (challenge 2): ", XmasCount)
#Right answer: 1854

File: 2024/dec04/C/02.c
#include "dec04.h"

int main() {
    FILE *file;
    char input[ROWS][COLUMNS] = {0};
    int r=0,c=0;
    char ch;

    file=fopen("../input.txt","r");
    if (file==NULL) return 1;
    
    //prepare 2D array
    while ((ch = fgetc(file)) != EOF && r<ROWS) {
        if (ch == '\n') {
            r++;
            c=0;
        }
        else {
            input[r][c]=ch;
            c++;
        }
    }

    int XmasCount=0;

    for (int linh=0; linh < ROWS; linh++)
        for (int coln=0; coln < COLUMNS; coln++)
            if (x_mas(input, linh, coln)) XmasCount++;

    printf("X-MAS Count (challenge 2): %d\n", XmasCount);
    //Right answer: 1854
    
}